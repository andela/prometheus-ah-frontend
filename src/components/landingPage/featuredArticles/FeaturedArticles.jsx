import React from 'react';
import PropTypes from 'prop-types';
import FeaturedArticle from './FeaturedArticle';

const FeaturedArticles = (props) => {
  const { posts } = props;
  return (
    <div className="col-12 col-md-7 col-lg-7">
      <h2 className="section-title">FEATURED POSTS</h2>
      {
        posts.map((post, index) => (
          <FeaturedArticle
            key={post.id}
            active={index}
            image={post.image}
            title={post.title}
            date={post.createdAt}
            category={post.Tags[0]}
            description={post.description}
            readTime={post.readingTime}
            author={`${post.User.firstname} ${post.User.lastname}`}
          />
        ))
      }
    </div>
  );
};

FeaturedArticles.propTypes = {
  posts: PropTypes.instanceOf(Array)
};

export default FeaturedArticles;
