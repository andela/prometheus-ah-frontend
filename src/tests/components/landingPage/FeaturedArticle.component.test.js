import React from 'react';
import { shallow } from 'enzyme';
import FeaturedArticles from '../../../components/landingPage/featuredArticles/FeaturedArticle';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('FeaturedArticles component', () => {
  beforeEach(() => {
    const {
      active, image, title, date, category, description, readTime
    } = mockData.postProps;

    const props = {
      active,
      image,
      title,
      date,
      category,
      description,
      readTime,
      author: {
        username: 'faksam',
        firstname: null,
        lastname: null
      }
    };

    wrapper = shallow(<FeaturedArticles {...props} />);
  });

  test('should render a snapshot of the FeaturedArticles component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
