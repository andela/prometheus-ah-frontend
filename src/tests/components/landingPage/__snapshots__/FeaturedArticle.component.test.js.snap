// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedArticles component should render a snapshot of the FeaturedArticles component 1`] = `
<div
  className="card shadow-lg border-0 mb-5 featured-post"
>
  <img
    alt=""
    className="img-fluid featured-post-image"
    src="https://image.ibb.co/eLVQwL/kids.jpg"
  />
  <div
    className="card-body"
  >
    <Link
      className="card-title post-title"
      replace={false}
      to="/articles/undefined"
    >
      My Broken Mother, My Broken Heart
    </Link>
    <p
      className="card-text post-date"
    >
      Sep 20, 2018
      <span
        className="ml-4 font-weight-light font-italic text-uppercase"
      >
        Technology
      </span>
    </p>
    <p
      className="card-text post-description"
    >
      feeling very excited
    </p>
    <div
      className="d-flex justify-content-between"
    >
      <div
        className="card-text post-author"
      >
        <p
          className="author"
        >
          faksam
        </p>
        <p>
          <small
            className="text-muted font-italic"
          >
            1 min read
          </small>
        </p>
      </div>
      <i
        className="far fa-bookmark"
      />
    </div>
  </div>
</div>
`;
